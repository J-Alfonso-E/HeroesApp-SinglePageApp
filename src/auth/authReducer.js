import { types } from '../types/types';

/*const state = {
    name: 'Poncho', 
    logged: true,
};*/

export const authReducer = (state = {}, action) => {
    //console.log(state);
    //console.log(action);
    //console.log(types.logout);
    switch ( action.type ) {
        case types.login:
            return {
                ...action.payload,
                logged: true
            }

        case types.logout:
            return {
                logged: false
            }
    
        default:
            return state;
    }
}