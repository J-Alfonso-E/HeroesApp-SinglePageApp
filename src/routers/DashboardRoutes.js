import { Routes, Route } from "react-router-dom";
import { MarvelScreen } from "../components/marvel/MarvelScreen";
import { SearchScreen } from "../components/search/SeachScreen";
import { DCScreen } from "../components/dc/DCScreen";
import { Navbar } from "../components/ui/Navbar";
import { HeroesScreen } from "../components/Heroes/HeroesScreen";

export const DashboardRoutes = () => {
    return (
        <>
            <Navbar />
            <div className="container">
                <Routes>

                    <Route path="marvel" element={<MarvelScreen />} />
                    <Route path="dc" element={<DCScreen />} />
                    <Route path="search" element={<SearchScreen />} />
                    {/* Ruta para pantalla de Heroe*/}
                    <Route path="hero/:HeroId" element={<HeroesScreen />} /> {/* Los 2 puntos y un nombre sive para decir que es una varable y se puede tener la informacion en donde se renderiza*/}
                    <Route path="/" element={<MarvelScreen />} />
                </Routes>
            </div>
        </>
    )
}