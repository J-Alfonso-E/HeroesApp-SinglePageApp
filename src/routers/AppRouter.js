import * as React from "react";
import { Routes, Route, Link, BrowserRouter } from "react-router-dom";
import { LoginScreen } from "../components/Login/loginScreen";

import { DashboardRoutes } from "./DashboardRoutes";


export const AppRouter = () => {
    return (
        <BrowserRouter>
            
            <Routes>
                <Route path="/login" element={<LoginScreen />} />
                <Route path="/*" element={<DashboardRoutes />} />
            </Routes>
        </BrowserRouter>
    );
}