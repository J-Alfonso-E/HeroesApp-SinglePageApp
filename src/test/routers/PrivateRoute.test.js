import {mount} from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { AuthContext } from '../../auth/authContext';
import {PrivateRoute} from '../../routers/PrivateRoute';

jest.mock('react-router-dom', () => ({ //Simulacion de la libreria de de react-router-dom
    ...jest.requireActual('react-router-dom'), //Importa la libreria de react sin cambios
    Navigate: () => <span>No Autenticado</span> //Despues de que se importo se sobreescribe el hook de Navigate
}));

Storage.prototype.setItem = jest.fn();


describe('Pruebas en PrivateRoute', ()=>{ 

    

    test('debe de mostrar el componenete si esta autenticado y tambien debe estar guardado en el localStorage', () => {

        const contextValue = {
            user: {
                logged: true, 
                name: 'Poncho'
            }
        };

        const wrapper = mount(
            <AuthContext.Provider value={contextValue} >
                <MemoryRouter initialEntries={["/"]} >
                    <PrivateRoute>
                        <h1>Private Component</h1>
                    </PrivateRoute>
                </MemoryRouter>
            </AuthContext.Provider>
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.text().trim()).toBe('Private Component');
        expect(localStorage.setItem).toHaveBeenCalledWith('lastpath', '/');

    });

    test('Debe mostrar algo diferente cuando no esta autenticado', () => {

        const contextValue = {
            user: {
                logged: false
            }
        };

        const wrapper = mount(
            <AuthContext.Provider value={contextValue} >
                <MemoryRouter initialEntries={["/"]} >
                    <PrivateRoute>
                        <h1>Private Component</h1>
                    </PrivateRoute>
                </MemoryRouter>
            </AuthContext.Provider>
        );

        
        expect(wrapper.text().trim()).toBe('No Autenticado');
        

    });

});
