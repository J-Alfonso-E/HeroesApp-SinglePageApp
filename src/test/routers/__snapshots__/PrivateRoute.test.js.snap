// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en PrivateRoute debe de mostrar el componenete si esta autenticado y tambien debe estar guardado en el localStorage 1`] = `
<h1>
  Private Component
</h1>
`;
